kornia/augmentation/_2d/mix/base.py

```diff
@@ -104,3 +108,162 @@ def forward(  # type: ignore
         if in_trans is not None:
             return (output, in_trans), lab
         return output, lab
+
+
+class MixAugmentationBaseV2(_BasicAugmentationBase):
+    r"""MixAugmentationBase base class for customized mix augmentation implementations.
+
+    For any augmentation, the implementation of "generate_parameters" and "apply_transform" are required.
+    "apply_transform" will need to handle the probabilities internally.
+
+    Args:
+        p: probability for applying an augmentation. This param controls if to apply the augmentation for the batch.
+        p_batch: probability for applying an augmentation to a batch. This param controls the augmentation
+          probabilities batch-wise.
+        same_on_batch: apply the same transformation across the batch.
+        keepdim: whether to keep the output shape the same as input ``True`` or broadcast it
+          to the batch form ``False``.
+    """
+
+    def __init__(
+        self, p: float, p_batch: float, same_on_batch: bool = False, keepdim: bool = False,
+        data_keys: List[Union[str, int, DataKey]] = [DataKey.INPUT],
+    ) -> None:
+        super().__init__(p, p_batch=p_batch, same_on_batch=same_on_batch, keepdim=keepdim)
+        self.data_keys = [DataKey.get(inp) for inp in data_keys]
+
+    def transform_tensor(self, input: Tensor) -> Tensor:
+        """Convert any incoming (H, W), (C, H, W) and (B, C, H, W) into (B, C, H, W)."""
+        _validate_input_dtype(input, accepted_dtypes=[torch.float16, torch.float32, torch.float64])
+        return _transform_input(input)
+
+    def apply_transform(  # type: ignore
+        self, input: Tensor, params: Dict[str, Tensor],
+        flags: Optional[Dict[str, Any]] = None
+    ) -> Tensor:
+        raise NotImplementedError
+
+    def apply_non_transform(  # type: ignore
+        self, input: Tensor, params: Dict[str, Tensor],
+        flags: Optional[Dict[str, Any]] = None
+    ) -> Tensor:
+        # For the images where batch_prob == False.
+        return input
+
+    def apply_func(  # type: ignore
+        self, in_tensor: Tensor, params: Dict[str, Tensor],
+        flags: Optional[Dict[str, Any]] = None
+    ) -> Tensor:
+        if flags is None:
+            flags = self.flags
+        to_apply = params['batch_prob']
+
+        # if no augmentation needed
```

Comment:
```suggestion
        # if no augmentation needed
        maybe_apply = to_apply.sum()
        if maybe_apply == 0:
```
to avoid computing sum twice in the elif branch